vllm/model_executor/layers/quantization/utils/fp8_utils.py

```diff
@@ -198,20 +201,30 @@ class W8A8BlockFp8LinearOp:
 
     def __init__(
         self,
+        weight_group_shape: GroupShape,
+        act_quant_group_shape: GroupShape,
         cutlass_block_fp8_supported: bool = CUTLASS_BLOCK_FP8_SUPPORTED,
         use_aiter_and_is_supported: bool = False,
     ):
+        self.weight_group_shape = weight_group_shape
+        self.act_quant_group_shape = act_quant_group_shape
         self.is_deep_gemm_supported = is_deep_gemm_supported()
-        self.ue8m0_deepgemm_supported = is_deep_gemm_e8m0_used()
         self.is_hopper = current_platform.is_device_capability(90)
-        self.w8a8_blockscale_op = self._dispatch_w8a8_blockscale_op(
+        self.w8a8_blockscale_op, self.input_quant_op = \
+         self._dispatch_w8a8_blockscale_op(
             cutlass_block_fp8_supported, use_aiter_and_is_supported)
+        self.deepgemm_input_quant_op = (QuantFP8(
```

Comment:
Why not just pass the proper `use_ue8m0=is_deep_gemm_e8m0_used()` to the `_dispatch_w8a8_blockscale_op` method so that we can always use `self.input_quant_op`?

Reply:
Oh I see, because we don't know at init time what size the tensor will be and so we need separate quants for deepgemm and cutlass/triton paths.

Reply:
I'll also add a comment on this